utils/build-script-impl

```diff
@@ -1832,7 +1833,7 @@ for host in "${ALL_HOSTS[@]}"; do
                 #
                 # This makes it easier to build target stdlibs on systems that
                 # have old toolchains without more modern linker features.
-                if [[ "$(uname -s)" != "Darwin" ]] ; then
+                if [[ "$(uname -s)" != "Darwin" || "${BUILD_LLD}" ]] ; then
                     if [[ ! "${SKIP_BUILD_LLD}" ]]; then
```

Comment:
@ahoppen I was thinking... we already have a skip-build-lld option... so we really shouldn't add another build-lld option to build-script-impl. In general we want to simplify build-script-impl by not adding new state like this.

I have a counter-proposal, why don't we remove the Darwin check here. Instead, in build-script lets just pass in skip-build-lldb to build-script-impl if we are on Darwin and build-lld isn't set. Then build-script-impl is simpler and doesn't have an additional variable. What are your thoughts?